pennylane/_device.py

```diff
@@ -1007,6 +1011,21 @@ def check_validity(self, queue, observables):
                         raise DeviceError(
                             f"Observable {i.name} not supported on device {self.short_name}"
                         )
+
+            elif isinstance(o, qml.ops.Prod):
+
+                supports_prod = self.supports_observable(o.name)
+                if not supports_prod:
+                    raise DeviceError(f"Observable Prod not supported on device {self.short_name}")
+
+                simplified_op = o.simplify()
+                if isinstance(simplified_op, qml.ops.Prod):
+                    for i in o.simplify().operands:
+                        if not self.supports_observable(i.name):
+                            raise DeviceError(
+                                f"Observable {i.name} not supported on device {self.short_name}"
```

Comment:
```suggestion
                                f"Observable {i} of type {type(i)} not supported on device {self.short_name}"
```

Reply:
I felt that the original error message was quite clear so I didn't change anything.